Google.PowerShell/PubSub/GcpsSubscription.cs

```diff
@@ -153,4 +153,166 @@ protected override void ProcessRecord()
+                do
+                {
+                    ListTopicSubscriptionsResponse response = null;
+                    try
+                    {
+                        response = listRequest.Execute();
+                    }
+                    catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
+                    {
+                        WriteResourceMissingError(
+                            exceptionMessage: $"Topic '{Topic}' does not exist in project '{Project}'.",
+                            errorId: "TopicNotFound",
+                            targetObject: Topic);
+                    }
+
+                    if (response?.Subscriptions != null)
+                    {
+                        // If user gives us a list of subscriptions to search for, we have to make sure that
+                        // those subscriptions belong to the topic.
+                        if (Subscription != null)
+                        {
+                            IEnumerable<string> selectedSubscriptions = response.Subscriptions
+                                .Where(sub => Subscription.Contains(sub, System.StringComparer.OrdinalIgnoreCase));
+                            GetSubscriptions(selectedSubscriptions);
+                        }
+                        else
+                        {
+                            GetSubscriptions(response.Subscriptions);
+                        }
+                    }
+                    listRequest.PageToken = response.NextPageToken;
+                }
+                while (!Stopping && listRequest.PageToken != null);
+                return;
+            }
+
+            // If no topic is given, then we are left with 2 cases:
+            // 1. User gives us a list of subscriptions: we just find those subscriptions and returned.
+            // 2. User does not give us any subscriptions: we just return all subscriptions in the project.
+            if (Subscription != null && Subscription.Length > 0)
+            {
+                GetSubscriptions(Subscription.Select(item => GetProjectPrefixForSubscription(item, Project)));
+            }
+            else
+            {
+                ProjectsResource.SubscriptionsResource.ListRequest listRequest =
+                    Service.Projects.Subscriptions.List($"projects/{Project}");
+                do
+                {
+                    ListSubscriptionsResponse response = listRequest.Execute();
+
+                    if (response.Subscriptions != null)
+                    {
+                        WriteObject(response.Subscriptions, true);
+                    }
+                    listRequest.PageToken = response.NextPageToken;
+                }
+                while (!Stopping && listRequest.PageToken != null);
+            }
+        }
+
+        /// <summary>
+        /// Given a list of subscription names, returns the corresponding subscriptions.
```

Comment:
" returns the corresponding subscriptions." Should this be "writes the corresponding subscriptions."?